client/clientBean.go

```diff
@@ -45,6 +46,7 @@ type (
 		GetHistoryClient() history.Client
 		GetMatchingClient() matching.Client
 		GetFrontendClient() frontend.Client
+		GetPublicClient() public.Client
```

Comment:
FrontendClient and PublicClient are same thing.  Why do we need separate definitions?

Reply:
because each time the idl on the frontend is updated, developer will have to update the public client first, then get it approved, then update the public sever repo pointing to public client, even for developing features.

Reply:
tried to use client IDL (revert back to prev impl: frontend depends on public client idl)
this will break the redirection handler: although the definition contents are exactly the same, the package name are different. this leads to only 2 solution: either change the idl definition dependency of frontend handler, which will break a lot of code, or use the solution in this PR

created a issue #1414, so in the future, client idl and service idl will be in the same repo